src/vsphere_cpi/spec/integration/vm_existing_disk.rb

```diff
@@ -0,0 +1,84 @@
+require 'integration/spec_helper'
+
+describe 'Creating VM with existing persistent disks' do
+  before (:all) do
+    @datacenter_name = fetch_and_verify_datacenter('BOSH_VSPHERE_CPI_DATACENTER')
+    @cluster_name = fetch_and_verify_cluster('BOSH_VSPHERE_CPI_CLUSTER')
+    @datastore_shared_pattern = fetch_and_verify_datastore('BOSH_VSPHERE_CPI_SHARED_DATASTORE', @cluster_name)
+    @cluster_one_datastore = fetch_and_verify_datastore('BOSH_VSPHERE_CPI_DATASTORE_PATTERN', @cluster_name)
+  end
+
+  context 'when there are existing persistent disks ' do
+    let(:vm_type) do
+      {
+        'ram' => 512,
+        'disk' => 2048,
+        'cpu' => 1
+      }
+    end
+    let(:cpi) do
+      options = cpi_options(
+        'datacenters' => [
+          {
+            'name' => @datacenter_name,
+            'datastore_pattern' => @datastore_shared_pattern,
+            'persistent_datastore_pattern' => @datastore_shared_pattern,
+            'clusters' => [
+              {
+                @cluster_name => {}
+              },
+            ]
+          }
+        ]
+      )
+      VSphereCloud::Cloud.new(options)
+    end
+
+    let(:cpi_different_persistent_pattern) do
+      options = cpi_options(
+        'datacenters' => [
+          {
+            'name' => @datacenter_name,
+            'datastore_pattern' => @datastore_shared_pattern,
+            'persistent_datastore_pattern' => @cluster_one_datastore,
+            'clusters' => [
+              {
+                @cluster_name => {}
+              },
+            ]
+          }
+        ]
+      )
+      VSphereCloud::Cloud.new(options)
+    end
+
+    before do
+      @disk_id_one = cpi.create_disk(2048, {}, nil)
+      @disk_id_two = cpi.create_disk(2048, {}, nil)
+      expect(@disk_id_one).to_not be_nil
+      expect(@disk_id_two).to_not be_nil
+    end
+
+    it 'cpi should be able to create a VM' do
+      begin
+        vm_id = cpi_different_persistent_pattern.create_vm(
+          'agent-007',
+          @stemcell_id,
+          vm_type,
+          get_network_spec,
+          [@disk_id_one, @disk_id_two],
+          {}
+        )
+        expect(vm_id).to_not be_nil
+        expect(cpi_different_persistent_pattern.has_vm?(vm_id)).to be(true)
```

Comment:
you could add another expectation here that the disk is in new datastore

Reply:
Disk not in new datastore. it will be when `attach_disk` is called. That never happens now as we are in integration test, but BOSH will likely do it